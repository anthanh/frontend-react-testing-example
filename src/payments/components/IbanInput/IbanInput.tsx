import React, { Component, SyntheticEvent, ChangeEvent } from "react";
import TextField from "@material-ui/core/TextField";
import { validateIbanFormat } from "../../helpers/payment.helpers";

interface IbanInputProps {
  value?: string;
  valid?: boolean;
}

class IbanInput extends Component<IbanInputProps, {}> {
  state: IbanInputProps;

  constructor(props: IbanInputProps) {
    super(props);
    this.state = {
      value: "",
      valid: true
    };
  }

  setValue = (event: ChangeEvent<HTMLInputElement>) => {
    this.setState({
      value: event.target.value,
      valid: validateIbanFormat(event.target.value)
    });
  }

  render() {
    return (
      <div>
        <label htmlFor="iban">IBAN</label>
        <TextField
          id="iban"
          placeholder="ES12 1234 1234 1234 1234"
          onChange={this.setValue}
          value={this.state.value}
        />
        <p className="iban-value">{this.state.value}</p>
        <p className="iban-valid">{JSON.stringify(this.state.valid)}</p>
      </div>
    );
  }
}

export default IbanInput;
