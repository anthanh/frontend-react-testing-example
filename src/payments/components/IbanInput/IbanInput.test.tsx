import React from "react";
import { shallow, mount } from "enzyme";
import IbanInput from ".";

describe("IbanInput", () => {
  let rendered;
  it("renders as expected", () => {
    rendered = shallow(<IbanInput />);
    expect(rendered.html()).toMatchSnapshot();
    expect(rendered.contains(<label htmlFor="iban">IBAN</label>)).toBe(true);
    expect(rendered.find("TextField")).toHaveLength(1);
  });

  it("renders as expected", () => {
    rendered = mount(<IbanInput />);

    expect(rendered.find("input")).toHaveLength(1);

    // rendered
    //   .find("IbanInput")
    //   .instance()
    //   .setValue({ target: { value: "customvalue" } });

    // rendered.update();
    // expect(rendered.find("input").props().value).toBe("customvalue");

    rendered
      .find("input")
      .simulate("change", { target: { value: "customvalue" } });
    expect(rendered.find(".iban-value").html()).toBe(
      '<p class="iban-value">customvalue</p>'
    );

    rendered
      .find("input")
      .simulate("change", { target: { value: "AT61 1904 3002 3457 1234" } });
    expect(rendered.find(".iban-valid").html()).toBe(
      '<p class="iban-valid">true</p>'
    );
  });
});
