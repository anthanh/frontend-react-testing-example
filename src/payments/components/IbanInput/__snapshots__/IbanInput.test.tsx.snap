// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IbanInput renders as expected 1`] = `"<div><label for=\\"iban\\">IBAN</label><div class=\\"MuiFormControl-root-1\\"><div class=\\"MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-formControl-19 MuiInput-formControl-6\\"><input type=\\"text\\" aria-invalid=\\"false\\" class=\\"MuiInputBase-input-28 MuiInput-input-13\\" id=\\"iban\\" placeholder=\\"ES12 1234 1234 1234 1234\\" value=\\"\\"/></div></div><p class=\\"iban-value\\"></p><p class=\\"iban-valid\\">true</p></div>"`;
